ethcore/types/src/receipt.rs

```diff
@@ -125,6 +125,8 @@ pub struct RichReceipt {
 	pub log_bloom: LogBloom,
 	/// State root
 	pub state_root: Option<H256>,
+	/// Status byte. Optional before EIP-658.
+	pub status_code: Option<u8>,
```

Comment:
since state root and status code are an either-or kind of thing, maybe it should be encoded as an enum instead of two separate fields? like
```rust
enum TransactionOutcome {
    Unknown(Option<H256>), // optional state root because of eip98
    Success,
    Failure,
}
```